src/clang-cpp-frontend/clang_cpp_adjust_expr.cpp

```diff
@@ -59,55 +59,49 @@ void clang_cpp_adjust::gen_implicit_union_copy_move_constructor(symbolt &symbol)
   ctor_body.operands().push_back(copy_ctor_assign);
 }
 
-void clang_cpp_adjust::gen_lambda_operator(symbolt &symbol)
+void clang_cpp_adjust::replace_lambda_capture_fields(
+  const irept &lambda_capture_fields,
+  exprt &op)
 {
-  if (!symbol.value.need_lambda_init())
-    return;
-
-  code_typet &type = to_code_type(symbol.type);
-
-  if (type.return_type().id() == "constructor")
-    return;
-
-  code_blockt &body = to_code_block(to_code(symbol.value));
-
-  exprt this_ptr = symbol_exprt(
-    type.arguments()[0].get("#identifier"), type.arguments()[0].type());
-
-  const typet &class_symb = ns.follow(type.arguments()[0].type().subtype());
-
-  const struct_typet::componentst &components =
-    to_struct_type(class_symb).components();
-
-  for (const auto &c : components)
+  if (op.is_symbol())
   {
-    if (c.get_bool("#capture_this"))
+    irept replacement_symbol = lambda_capture_fields.find(op.identifier());
+    if (replacement_symbol.is_not_nil())
     {
-      // the clang declares the original variable (this->data)
-      // releace 'this->data' with 'this->__this->data'
-      // which this = this ptr of lambda and
-      // __this = capture this
-      exprt deref = dereference_exprt(this_ptr, this_ptr.type());
-      exprt dest = member_exprt(deref, c.get_name(), c.type());
-      exprt memderef =
-        dest.type().is_pointer() ? dereference_exprt(dest, dest.type()) : dest;
-
-      replace_capture_this(memderef, body);
+      op.swap(replacement_symbol);
+    }
+  }
+  else
+  {
```

Comment:
same here

Reply:
I think it would look odd to have braces for the `if`, but not for the `else`.